lib/parser.ypp

```diff
@@ -1261,6 +1261,33 @@ static AbstractQoreNode* processCall(int sline, int eline, AbstractQoreNode* $1,
    }
    return $$;
 }
+
+static void check_operator(const AbstractQoreNode *n, const char *parent, const char *child) {
+    if (n->getType() == NT_OPERATOR && parse_check_parse_option(PO_BROKEN_LOGIC_PRECEDENCE)) {
+        if (!dynamic_cast<const QoreOperatorNode *>(n)->getInParenthesis()) {
```

Comment:
do not use `dynamic_cast<>` here; it has unnecessary runtime overhead (and if it failed it would return `nullptr` anyway); use `static_cast<>` instead.   when I recommended `dynamic_cast<>` in an email I thought you just had a `QoreOperatorNode` at runtime and wanted to test for a subclass